uvtools/plot.py

```diff
@@ -2,115 +2,385 @@
+          - 'imag': Imaginary value.
+          - 'log':  Log (base-10) of absolute value.
+        Default: 'log'.
+    
+    vmin, vmax : float, optional
+        Minimum and maximum values of the color scale. If not set (and `mx` and 
+        `drng` are not set), the min. and max. values of the data will be used.
+        
+        Note that that the min. and max. values are the ones _after_ the data 
+        have been transformed according to `mode`. So, if `mode='log'`, these 
+        values are the min. and max. of log_10(data).
+        
+    mx : float, optional (deprecated)
+        The max. value of the color scale in the plot (equivalent to vmax). 
+        Cannot be specified at the same time as `vmin` and `vmax`.
+    
+    drng : float, optional (deprecated)
+        The difference between the min. and max. values of the color scale in 
+        the plot, `drng = mx - min`, where these are the min/max values after 
+        applying the transformation specified by `mode`.
+        
+        Cannot be specified at the same time as `vmin` and `vmax`.
+    
+    recenter : bool, optional
+        Recenter the image data by shifting by 50% around a circle (useful for 
+        recentering Fourier-transformed data). Default: False.
+    
+    Returns
+    -------
+    plot : matplotlib.imshow
+        Waterfall plot.
+    """
+    # Check validity of inputs
+    validity_msg = "Must specify either `vmin` and `vmax` *or* `mx` and `drng`."
+    if mx is not None or drng is not None:
+        assert vmin is None and vmax is None, validity_msg
+        
+    if vmin is not None or vmax is not None:
+        assert mx is None and drng is None, validity_msg
+        mx = vmax
+        drng = vmax - vmin
+    
     # import matplotlib
     import pylab as plt
-    if np.ma.isMaskedArray(d): d = d.filled(0)
-    if recenter: d = aipy.img.recenter(d, np.array(d.shape)/2)
+    
+    # Fill masked array and recenter if requested
+    if np.ma.isMaskedArray(d):
+        d = d.filled(0)
+    if recenter:
+        d = aipy.img.recenter(d, np.array(d.shape)/2)
+    
+    # Apply requested transform to data
     d = data_mode(d, mode=mode)
-    if mx is None: mx = d.max()
-    if drng is None: drng = mx - d.min()
+    
+    # Get min/max values for color scale
+    if mx is None:
+        mx = d.max()
+    if drng is None:
+        drng = mx - d.min()
     mn = mx - drng
-    if 'aspect' not in kwargs.keys(): kwargs['aspect'] = 'auto'
+    
+    # Choose aspect ratio
+    if 'aspect' not in kwargs.keys():
+        kwargs['aspect'] = 'auto'
     return plt.imshow(d, vmax=mx, vmin=mn, interpolation='nearest', **kwargs)
 
+
 def plot_hmap_ortho(h, cmap='jet', mode='log', mx=None, drng=None, 
-        res=0.25, verbose=False, normalize=False):
+                    res=0.25, verbose=False, normalize=False):
+    """
+    Plot a Healpix map in ortho projection.
+    
+    Parameters
+    ----------
+    h : healpix map
+        Healpy HEALPIX map.
```

Comment:
Could you specify further what is expected here? Is this a specific object from healpy? On line 201 it looks like it's some kind of interpolation function.

Reply:
On closer inspection, it turns out to be an aipy `HealpixMap` instance! I've changed the string.